students/anbaopham/lesson06/test_Assignment_4.py

```diff
@@ -0,0 +1,255 @@
+    return first_last_name_list
+
+def send_all_thank_you():
+    first_last_name_list= get_first_last_name()
+
+    for i in range(len(first_last_name_list)):
+        first_name = first_last_name_list[i][0]
+        last_name = first_last_name_list[i][1]
+        thank_you_letter_template(first_name, last_name, 10)
+
+def test_send_all_thank_you():
+    send_all_thank_you()
+    first_last_name_list= get_first_last_name()
+    path = os.getcwd()
+    for i in range(len(first_last_name_list)):
+        first_name = first_last_name_list[i][0]
+        last_name = first_last_name_list[i][1]
+        file_name = f"{first_name}_{last_name}.txt"
+    for i in os.listdir(path):
+        if i.startswith(file_name) and i.endswith(".txt"):
+            with open("file_name", "r") as fp:
+                f = fp.read()
+                assert(f) == template
+
+def sort_total_given(l1):
+    #return donor_db[0].split(" ")[1]
+    return l1[1]
+
+
+def get_report_data():
+    temp_list = []
+    for x in donor_db:
+        temp_list.append(x[0])
+        tg = round(sum(list((x[1]))),2)
+        total_given = str(tg)
+        temp_list.append(total_given)
+        ng = len(x[1])
+        num_gift = str(ng)
+        temp_list.append(num_gift)
+        aG = round(tg/ng,2)
+        ave_gift = str(aG)
+        temp_list.append(ave_gift)
+
+    return temp_list
+
+def test_get_report_data():
+    add_donor("an pham", 1234567)
+    donor_list = get_report_data()
+    assert(donor_list[-4]) == "an pham"
+    assert(donor_list[-3]) == '1234567'
+    assert(donor_list[-2]) == '1'
+    assert(donor_list[-1]) == '1234567.0'
+
+def create_a_report():
+    l = (["Donor Name", "Total Given", "Num Gifts", "Average Gift"])
+    temp_list = get_report_data()
+    l1 = []
+
+    m = 0
+    while m< len(temp_list):
+        l1.append(temp_list[m:m+4])
+        m = m+4
+
+    l1 =sorted(l1, key = sort_total_given)
+    l1.insert(0,l)
+
+    new_list = []
+
+    [new_list.append(len(j)) for i in l1 for j in i]
+    l = max(new_list)
+    space = " "*10
+    width = l
+    print(l1)
+
+    for a, b, c, d in l1:
+        print(f'{a:<{width}}{space}{b:>{width}}{space}{c:>{width}}{space}{d:>{width}}')
+
+def exit_program():
+    print("Bye!")
+    sys.exit()  # exit the interactive script
+
+response_dict ={"1": send_a_thank_you, "2": create_a_report, "3": send_all_thank_you, "4": exit_program}
+def main():
+    while True:
+        response = input(prompt)  # continuously collect user selection
+        # now redirect to feature functions based on the user selection
+        if response in response_dict:
+            response_dict.get(response)()
+        else:
+            print("Not a valid option!")
+
+        #try:
+        #    response_dict.get(response)()
+        #except:
+        #    print("Not a valide option!")
+
+
+
+
+if __name__ == "__main__":
+    # don't forget this block to guard against your code running automatically if this module is imported
+
+    main()
```

Comment:
There should be a separate file with unit tests for this assignment.  Do you have it?